onnx/defs/math/defs.cc

```diff
@@ -82,7 +83,29 @@ will throw errors.
         "T",
         {"tensor(float16)", "tensor(float)", "tensor(double)"},
         "Constrain input and output types to float tensors.");
-    schema.TypeAndShapeInferenceFunction(propagateShapeAndTypeFromFirstInput);
+    schema.TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+      propagateElemTypeFromInputToOutput(ctx, 0, 0);
+      if (!hasNInputShapes(ctx, 1)) {
+        return;
+      }
+      propagateShapeFromInputToOutput(ctx, 0, 0);
+      const TensorShapeProto& input_shape =
+        ctx.getInputType(0)->tensor_type().shape();
+      int r = input_shape.dim_size();
+      if (r != 2) {
```

Comment:
I think this check does not follow the definition of `Softmax`? The document of `Softmax` says "Input does not need to explicitly be a 2D vector; rather, it will be coerced into one".